Dockerfile

```diff
@@ -1,38 +1,107 @@
+#     appears to be buggy and doesn't actually function in this case, resulting
+#     in copies of all the libraries being redownloaded from the internet.
+#
+#  2. The installation installs via rsync archive, which will keep both copies
+#     around, resulting in an extra ~50MB of duplicates in the image.
+#
+# Therefore, we could relocate the preload directory to where it should end
+# up (usually $HOME/.sbt/preloaded, but see getPreloaded in sbt-launch.lib.bash
+# for more details of how this is calculated) instead of leaving two copies
+# lying around.
+#
+# HOWEVER, even then, the first execution of sbt will "download" from this
+# preload folder, creating a third copy with a slightly different directory
+# structure in $HOME/.ivy2/cache (yep, it copies, it doesnt symlink or move).
+# And naturally that directory structure is different enough that you can't just
+# put the local-preloaded directory there to begin with. Thus, we need to
+# execute sbt once and then delete the preload cache after.
+#
+# Additionally, there are Windows specific files included in the download that
+# we remove to save space and avoid confusion (bin/sbt.bat conf/sbtconfig.txt).
+#
+# And yep, we do this all in one mega command to keep the layer small. If you
+# are working on this in the future, https://github.com/wagoodman/dive is your
+# friend.
+#
+# NOTE: there is currently an experimental sbt pkg for alpine in edge/testing:
+#
+#     https://git.alpinelinux.org/aports/tree/testing/sbt/APKBUILD.
+#
+# But we don't want to depend on something outside of the stable alpine
+# tracking, and this additionally does not handle the installation of the
+# local-preloaded library -- so if switching to it in the future be sure to
+# cache that directory manually as well.
+RUN apk add --no-cache --virtual=build-deps curl && \
+    # Install sbt base
+    mkdir -p "${SBT_HOME}" && \
+    set -o pipefail && \
+    curl -fsL "https://github.com/sbt/sbt/releases/download/v${SBT_VERSION}/sbt-${SBT_VERSION}.tgz" \
+      | tar xfz - --strip-components=1 -C "${SBT_HOME}" && \
+    ln -s "${SBT_HOME}"/bin/sbt /usr/local/bin/sbt && \
+    # Put the preloaded library files where sbt will detect them, and let it
+    # invoke once which will "download" them from local filesystem
+    mkdir -p "${HOME}"/.sbt && \
+    mv "${SBT_HOME}"/lib/local-preloaded "${HOME}"/.sbt/preloaded && \
+    sbt ++"${SCALA_VERSION}" sbtVersion && \
+    # Now that sbt is happy, get rid of the preload library directory so
+    # we dont have two copies taking up extra space (its about 50mb!)
+    rm -r "${HOME}"/.sbt/preloaded && \
+    # Get rid of Windows specific files
+    rm "${SBT_HOME}"/bin/sbt.bat && \
+    rm "${SBT_HOME}"/conf/sbtconfig.txt && \
+    # Get rid of build-dependencies we only needed for this install step
+    apk del build-deps
+
+# Verify SBT is installed successfully. This layer also exists to ensure it is a
```

Comment:
I feel like there might be a clearer way to phrase this

Reply:
Tried rephrasing.